tortoisebeacon/tortoise_beacon.go

```diff
@@ -326,7 +318,14 @@ func (tb *TortoiseBeacon) cleanup() {
 	for e := range tb.beacons {
 		// TODO(nkryuchkov): https://github.com/spacemeshos/go-spacemesh/pull/2267#discussion_r662255874
 		if tb.epochIsOutdated(e) {
+			delete(tb.validProposals, e)
+			delete(tb.potentiallyValidProposals, e)
 			delete(tb.beacons, e)
+			delete(tb.proposalPhaseFinishedTimestamps, e)
+			delete(tb.incomingVotes, e)
+			delete(tb.ownVotes, e)
+			delete(tb.firstRoundIncomingVotes, e)
+			delete(tb.firstRoundOutcomingVotes, e)
```

Comment:
can we release all this data immediately when the beacon for the epoch was calculated? precisely at the end of handleEpoch method, maybe even just a deferred statement to cleanup it even if we exit early.

i don't see why we need to store all this stuff in memory for 1000 epochs, with an exception of the beacon itself ofcourse (but should we store it in database actually?).

Reply:
whenever a beacon is calculated it needs to be stored in DB and all votes can be discarded at once

Reply:
Yes, we can release data immediately and store the beacon in the DB only. I also want to remove some internal `map`s. I'm working on that now, but I was going to have this in another PR because I wanted to keep the scope of this PR as small as possible to have it ready ASAP and merged before the devnet relaunch. The current implementation should reduce the memory usage by a lot and it introduces a few changes, which are easy to test because they are similar to already implemented beacon cleanup.